### ISSUE TITLE
Use same URL for all tenants and redirect after login ?

### ISSUE THREAD

**Issue by gabrielpasv** [django-tenants/django-tenants#935]
I wonder if there's a way to make users login to their tenant using one unique URL instead of tenant.domain.com

What I want is the user to login to domain.com and then at the login time it will detect their tenant and redirect it to the correct tenant.

not sure if it's possible

**Comment by bgunebakan** [comment]
Yes, it's possible. You can deploy authentication server or If you want to keep everything within a single Django project, you can do implementation of the middleware and authentication backend:

```python
# Middleware
class SetTenantMiddleware(MiddlewareMixin):
    def process_request(self, request):
        if request.user.is_authenticated:
            tenant_id = request.session.get('tenant_id')
            if tenant_id:
                request.tenant = Tenant.objects.get(id=tenant_id)
                connection.set_tenant(request.tenant)

# Authentication Backend
class TenantBackend(ModelBackend):
    def authenticate(self, request, username=None, password=None, tenant_id=None):
        user = super().authenticate(request, username=username, password=password)
        if user is not None:
            request.session['tenant_id'] = tenant_id
        return user
```

And in your login view:

```python
def login_view(request):
    if request.method == 'POST':
        form = LoginForm(request.POST)
        if form.is_valid():
            username = form.cleaned_data.get('username')
            password = form.cleaned_data.get('password')
            tenant_id = form.cleaned_data.get('tenant_id')
            user = authenticate(request, username=username, password=password, tenant_id=tenant_id)
            if user is not None:
                login(request, user)
                return redirect('home')
    else:
        form = LoginForm()
    return render(request, 'login.html', {'form': form})
```

You'll need to adapt it to your specific use case.

**Comment by hogking** [comment]
@bgunebakan Hi bgunebakan, thanks for your answer. It looks like this solution may need to know what the tenant is when login. My problem is that when user login(use phone number as account), I don't know what the tenant id is(because all tenants' user use same login page). My idea is to search from the users table of all tenants if the user exists first. Can you give me some suggestions? Thank you!

**Comment by tomturner** [comment]
Closing as part of a backlog tidy-up. This is a usage/integration question rather than a bug in django-tenants, and it has been inactive for some years — the versions and Django releases involved have moved on a long way since.

If you're still stuck, **[GitHub Discussions](https://github.com/django-tenants/django-tenants/discussions)** is the better home for questions like this and gets more eyes than the issue tracker.

And if you believe this *is* a bug in django-tenants, please reopen or open a fresh issue with the django-tenants and Django versions, a minimal `settings.py`, and the steps to reproduce — that's what makes it actionable.

Thanks for taking the time to write it up.
